packages/frontend-shared/src/graphql/urqlDetailedDebugExchange.ts

```diff
@@ -0,0 +1,65 @@
+import md5 from 'md5'
+import type { Exchange } from '@urql/core'
+import { pipe, tap } from 'wonka'
+import jsonStableStringify from 'json-stable-stringify'
+import type { OperationDefinitionNode } from 'graphql'
+import { print } from 'graphql'
+
+declare global {
+  interface Window {
+    __toPrintClear: Function
+    __toPrint__: {
+      queries: Record<any, any>
+      logs: any[]
+      data: Record<string, any>
+    }
+  }
+}
+
+window.__toPrintClear = function () {
+  window.__toPrint__ = {
+    queries: [],
+    logs: [],
+    data: {},
+  }
+}
+
+export const detailedDebugExchange: Exchange = ({ forward }) => {
```

Comment:
Seems this is not used?

Reply:
This was useful when debugging issues we were seeing around the urql cache not hitting properly, but it's extremely verbose. I can add a comment here that it can be imported/added to the middleware if you need to do advanced debugging of things in URQL req/response payloads